specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchservice.json

```diff
@@ -0,0 +1,6668 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "SearchServiceClient",
+    "description": "Client that can be used to manage and query indexes and documents, as well as manage other resources, on a search service.",
+    "version": "2019-05-06-Preview",
+    "x-ms-code-generation-settings": {
+      "useDateTimeOffset": true
+    }
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{endpoint}",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "$ref": "#/parameters/EndpointParameter"
+      }
+    ]
+  },
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/datasources('{dataSourceName}')": {
```

Comment:
I've not seen this format used for path parameters, does it actually work?  I think this is what the model validation is complaining about.  Usually the format is like this.
```json
/datasources/{dataSourceName}
```

Reply:
```
2020-02-28T03:28:13.3022135Z Validating "examples" and "x-ms-examples" in  specification/search/data-plane/Azure.Search/preview/2019-05-06-preview/searchservice.json:
2020-02-28T03:28:13.3023630Z 
2020-02-28T03:28:13.3044895Z [31m error : [0m
2020-02-28T03:28:13.3045792Z operationId: DataSources_CreateOrUpdate
2020-02-28T03:28:13.3046731Z scenario: SearchServiceCreateOrUpdateDataSource
2020-02-28T03:28:13.3047334Z source: request
2020-02-28T03:28:13.3047588Z responseCode: ALL
2020-02-28T03:28:13.3047879Z severity: 0
2020-02-28T03:28:13.3048114Z code: MISSING_REQUIRED_PARAMETER
2020-02-28T03:28:13.3048331Z details:
2020-02-28T03:28:13.3048622Z   code: MISSING_REQUIRED_PARAMETER
2020-02-28T03:28:13.3048925Z   message: Value is required but was not provided
2020-02-28T03:28:13.3050149Z   schemaPath: '#/paths/~1datasources(''{dataSourceName}'')/put/parameters/0'
```

Reply:
@jhendrixMSFT That format is part of the OData standard, and it works just fine (in the service anyway). It's a really good idea because otherwise, if a customer creates a resource called `indexes/prototype`, we can never add an API called `prototype` without breaking them.

The validation tools should be robust enough to handle different kinds of path syntax.

Reply:
Sorry I'm missing something here, the swagger specification doesn't mention anything about this format for path parameters.  So what gets replaced, the entire `('{dataSourceName}')`?

Reply:
No, just `{dataSourceName}`. The actual URL of a request would look like this:

```http
GET /datasources('mydatasource')
```

Our REST API would also accept

```http
GET /datasources/mydatasource
```

but we want client libraries to use the former syntax (OData) because it's less prone to ambiguity between user-defined resources and RPC-style APIs.

Reply:
Also, I reported this as a bug in the validator years ago, and I thought it was fixed. Maybe this is a regression?

Reply:
Ah ok, thanks for the explanation.
@NullMDR can you please take a look at the model validation failure regarding this?